app/assets/javascripts/components/client/EditClient.jsx

```diff
@@ -24,7 +82,190 @@ class EditClient extends React.Component {
+
+            <Input
+                type="checkbox"
+                title="Living With Parents"
+                placeholder="Living With Parents"
+                name="living_w_parents"
+                initData={client.living_w_parents}
+                update={this._update} />
+
+            <Input
+                type="number"
+                min="0"
+                title="Initial Intake"
+                placeholder="Initial Intake"
+                name="initial_intake"
+                initData={client.initial_intake}
+                update={this._update} />
+
+            <h3>Contact Information</h3>
+            <Input
+                type="text"
+                title="Phone Number"
+                placeholder="Phone Number"
+                name="phone_number"
+                initData={client.phone_number}
+                update={this._update} />
+
+            <Input
+                type="text"
+                title="Country"
+                placeholder="Country"
+                name="country"
+                initData={client.country}
+                update={this._update} />
+
+            <Input
+                type="text"
+                title="State"
+                placeholder="State"
+                name="state"
+                initData={client.state}
+                update={this._update} />
+
+            <Input
+                type="text"
+                title="Postal Code"
+                placeholder="Postal Code"
+                name="postal_code"
+                initData={client.postal_code}
+                update={this._update} />
+
+            <Input
+                type="text"
+                title="City"
+                placeholder="City"
+                name="city"
+                initData={client.city}
+                update={this._update} />
+
+            <Input
+                type="text"
+                title="Street"
+                placeholder="Street"
+                name="street"
+                initData={client.street}
+                update={this._update} />
+
+            <h3>Immigration History</h3>
+
+            <Input
+                type="date"
+                title="Court Date"
+                placeholder="Court Date"
+                name="court_date"
+                initData={client.court_date.slice(0, 10)}
+                update={this._update} />
+
+            <h4>Stage</h4>
+            <select
+                    className="select"
+                    name="stage"
+                    onChange={this._handleChange} 
+                    defaultValue={client.stage} >
+                    <option value="1">Case Opening</option>
+                    <option value="2">Case Starting</option>
+                    <option value="3">Middle Phase</option>
+                    <option value="4">Litigation</option>
+                    <option value="5">Post-Litigation</option>
+                    <option value="6">Case Closing</option>
```

Comment:
Is this stored as an enum on your backend? If so then you can think about pulling the possible values from the API and then getting the values from there as well so it's totally DRY

Reply:
ah, good idea!

Reply:
hm, it doesn't look like this is an enum. @amyshan do you think we should change this to be an enum?